xarray/core/types.py

```diff
@@ -84,14 +84,16 @@
         # anything with a dtype attribute
         _SupportsDType[np.dtype[Any]],
     ]
-    try:
-        from cftime import datetime as CFTimeDatetime
-    except ImportError:
-        CFTimeDatetime = Any
-    DatetimeLike = Union[pd.Timestamp, datetime.datetime, np.datetime64, CFTimeDatetime]
+
 else:
     DTypeLikeSave: Any = None
 
+try:
+    from cftime import datetime as CFTimeDatetime
+except ImportError:
+    CFTimeDatetime = Any
+DatetimeLike = Union[pd.Timestamp, datetime.datetime, np.datetime64, CFTimeDatetime]
```

Comment:
needed for `TimeResampler` dataclass

Reply:
If cftime is not installed, then `DatetimeLike = Any` and then strings are valid DateTimeLikes as well! 
Try moving the DateTimeLike inside the try/except?